rmgpy/cantherm/input.py

```diff
@@ -388,27 +411,111 @@ def loadInputFile(path):
+            job.cantherm_species.species_dictionary['atom_energies'] = atom_energies
     
     return jobList
+
+
+def load_species_from_database_file(job):
+    """
+    Load the species with all statMech data from the .yml file
+    """
+    global jobList
+    logging.info('Loading statistical mechanics parameters for {0} from .yml file...'.format(job.species.label))
+    yaml_registers()
+    with open(job.path, 'r') as f:
+        data = yaml.safe_load(stream=f)
+
+    try:
+        if job.species.label != data['label']:
+            logging.warning('Found different labels for species: {0} in input file, and {1} in .yml file. '
+                            'Using {0}.'.format(job.species.label, data['label']))
+    except KeyError:
+            logging.debug('Did not find label for species {0} in .yml file.'.format(job.species.label))
+
+    try:
+        job.species.conformer = data['conformer']
+    except KeyError:
+        raise ValueError('Could not determine Conformer for species {0} imported from .yml file'.format(
+            job.species.label))
+
+    try:
+        job.species.symmetryNumber = data['symmetryNumber']
+    except KeyError:
+        logging.debug('Could not determine symmetryNumber for species {0} imported from .yml file'.format(
+            job.species.label))
+
+    try:
+        job.species.transportData = data['transport_data']
+    except KeyError:
+        if is_pdep():
+            raise ValueError('Could not determine transportData for species {0} imported from .yml file'.format(
+                job.species.label))
+
+    try:
+        job.species.energyTransferModel = data['energy_transfer_model']
+    except KeyError:
+        if is_pdep():
+            raise ValueError('Could not determine energyTransferModel for species {0} imported from .yml file'.format(
+                job.species.label))
+
+    try:
+        job.species.molecularWeight = Quantity(data['molecularWeight'][0], data['molecularWeight'][1])
+    except KeyError:
+        if is_pdep():
+            job.species.molecularWeight = determine_molecular_weight(job.species)
+
+    mol = None
+    try:
+        mol = Molecule().fromAdjacencyList(data['adjacencyList'])
+    except InvalidAdjacencyListError:
+        try:
+            mol = Molecule(SMILES=data['SMILES'])
+        except ValueError:
+            try:
+                mol = Molecule().fromInChI(data['InChI'])
+            except ValueError:
+                try:
+                    number = np.ndarray(shape=(len(job.species.conformer.number.value_si)))
+                    number[:] = [num for num in job.species.conformer.number.value_si]
+                    coordinates = np.ndarray(shape=(len(number), 3))
+                    for i, coorlist in enumerate(job.species.conformer.coordinates.value_si):
+                        coordinates[i] = [coor for coor in coorlist]
+                    mol = Molecule().fromXYZ(atomicNums=number, coordinates=coordinates)
```

Comment:
This section seems oddly coded. Was it done intentionally? It seems like the following would be sufficient:
```
number = job.species.conformer.number.value_si.tolist()
coordinates = job.species.conformer.coordinates.value_si.tolist()
```

Reply:
Oh, I misinterpreted it, I thought initially that the conformer attributes were `np.ndarray` objects, and you were converting them into lists.

If the conformer attributes are `np.ndarray`, it seems you could use them directly. If they're lists, then couldn't you use `np.array(list_input)` to convert it to an `np.ndarray`?